finality-aleph/src/network/tcp.rs

```diff
@@ -207,6 +210,31 @@ pub async fn new_tcp_network<A: ToSocketAddrs>(
     Ok((TcpDialer {}, listener, identity))
 }
 
+pub async fn new_rate_limited_network<A: ToSocketAddrs>(
```

Comment:
Actually, you might want to disregard the comment below, and instead make a function constructing the rate limited `Dialer`/`Listener` pair out of a non-limited pair in the `rate_limiter` crate – this should be possible after moving the implementations from the `clique` crate as I mentioned above. Then you could just call that function, and even make it return `impl`s rather than concrete types, letting you un`pub` the rate limited dialer/listener structs. Yay, less API!

Reply:
Check it now. Now you need to use `RateLimitedDialer/Listener` by yourself (that function had annoying signature), `TokenBucket` is impl detail of `SleepingRateLimiter`.